src/hyperlink.rs

```diff
@@ -0,0 +1,87 @@
+use crate::filesystem::absolute_path;
+use std::fmt::{self, Formatter, Write};
+use std::path::{Path, PathBuf};
+use std::sync::OnceLock;
+
+pub(crate) struct PathUrl(PathBuf);
+
+#[cfg(unix)]
+static HOSTNAME: OnceLock<String> = OnceLock::new();
+
+impl PathUrl {
+    pub(crate) fn new(path: &Path) -> Option<PathUrl> {
+        Some(PathUrl(absolute_path(path).ok()?))
+    }
+}
+
+impl fmt::Display for PathUrl {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "file://{}", host())?;
```

Comment:
Why do we need the hostname?  Does `file:///home/user/...` not work?

Reply:
It does. As long as fd is running on the same host as the terminal. Including the hostname is useful if you run the command over ssh or in a container or VM. 

See https://gist.github.com/egmontkob/eb114294efbcd5adb1944c9f3cb5feda#file-uris-and-the-hostname